dotCMS/src/main/java/com/dotmarketing/portlets/workflows/business/WorkflowAPIImpl.java

```diff
@@ -1380,55 +1376,90 @@ public void updateStepReferences(String stepId, String replacementStepId) throws
      * @param user User to validate
      * @param respectFrontEndRoles indicates if should respect frontend roles
      * @param permissionable ContentType or contentlet to validate special workflow roles
-     * @param anyWhoCanViewContent Workflow action role
-     * @param anyWhoCanEditContent Workflow action role
-     * @param anyWhoCanPublishContent Workflow action role
-     * @param anyWhoCanEditPermisionsContent Workflow action role
      * @param action The action to validate
      * @return true if the user has one of the special workflow action role permissions, false if not
      * @throws DotDataException
      */
     @CloseDBIfOpened
     private boolean hasSpecialWorkflowPermission(User user, boolean respectFrontEndRoles,
-            Permissionable permissionable, Role anyWhoCanViewContent, Role anyWhoCanEditContent,
-            Role anyWhoCanPublishContent, Role anyWhoCanEditPermisionsContent,
-            WorkflowAction action) throws DotDataException {
+            Permissionable permissionable, WorkflowAction action) throws DotDataException {
+
+        Role anyWhoCanViewContent = roleAPI.loadRoleByKey(RoleAPI.WORKFLOW_ANY_WHO_CAN_VIEW_ROLE_KEY);
+        Role anyWhoCanEditContent = roleAPI.loadRoleByKey(RoleAPI.WORKFLOW_ANY_WHO_CAN_EDIT_ROLE_KEY);
+        Role anyWhoCanPublishContent = roleAPI.loadRoleByKey(RoleAPI.WORKFLOW_ANY_WHO_CAN_PUBLISH_ROLE_KEY);
+        Role anyWhoCanEditPermisionsContent = roleAPI.loadRoleByKey(RoleAPI.WORKFLOW_ANY_WHO_CAN_EDIT_PERMISSIONS_ROLE_KEY);
 
-    	if (APILocator.getPermissionAPI()
+    	if (permissionAPI
                 .doesRoleHavePermission(action, PermissionAPI.PERMISSION_USE,
                         anyWhoCanViewContent)) {
-            if (APILocator.getPermissionAPI()
+            if (permissionAPI
                     .doesUserHavePermission(permissionable, PermissionAPI.PERMISSION_READ,
                             user, respectFrontEndRoles)) {
                 return true;
             }
         }
-        if (APILocator.getPermissionAPI()
+        if (permissionAPI
                 .doesRoleHavePermission(action, PermissionAPI.PERMISSION_USE,
                         anyWhoCanEditContent)) {
-            if (APILocator.getPermissionAPI().doesUserHavePermission(permissionable,
+            if (permissionAPI.doesUserHavePermission(permissionable,

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1066)